misc/scripts/file_format.sh

```diff
@@ -29,22 +35,17 @@ while IFS= read -rd '' f; do
     recode UTF-8 "$f" 2> /dev/null
     # Ensures that files have LF line endings.
     dos2unix "$f" 2> /dev/null
-    # Ensures that files do not contain a BOM.
-    sed -i '1s/^\xEF\xBB\xBF//' "$f"
-    # Ensures that files end with newline characters.
-    tail -c1 < "$f" | read -r _ || echo >> "$f";
     # Remove trailing space characters.
-    sed -z -i 's/\x20\x0A/\x0A/g' "$f"
+    # -l option handles newlines conveniently and seems to also get rid of BOMs.
+    perl -i -ple 's/\s*$//g' "$f"
```

Comment:
I just tested this locally and it doesn't get rid of BOMs. Here's a test file which contains a BOM, after running `perl -i -ple 's/\s*$//g' test.txt`, the file still contains a BOM. [test.txt](https://github.com/godotengine/godot/files/4986397/test.txt)

Reply:
Well the script removed BOMs for me. I assumed it was this change, but maybe it's recode or dos2unix that does it.